### PR TITLE
Modify "End of service" text to specify routes and/or directions

### PR DESCRIPTION
**Pull request by sloanelybutsurely**
#### Summary of changes
**Asana Ticket:** [Modify "End of service" text to specify routes and/or directions](https://app.asana.com/0/1185117109217422/1207279074432055/f)

Updates Last Trip messages to add additional context w/r/t route or direction when possible.

#### Reviewer Checklist
- [x] Meets ticket's acceptance criteria
- [x] ~~Any new or changed functions have typespecs~~ No changes to functions
- [x] Tests were added for any new functionality (don't just rely on Codecov)
- [ ] This branch was deployed to the staging environment and is currently running with no unexpected increase in warnings, and no errors or crashes (compare on Splunk: [staging](https://mbta.splunkcloud.com/en-US/app/search/search?q=search%20index%3Drealtime-signs-dev%20%22%5Berror%5D%22%20OR%20%22%5Bwarn%5D%22%20OR%20%22CRASH%22&display.page.search.mode=verbose&dispatch.sample_ratio=1&earliest=-4h%40m&latest=now&sid=1545840107.3874236) vs. [prod](https://mbta.splunkcloud.com/en-US/app/search/search?q=search%20index%3Drealtime-signs-prod%20%22%5Berror%5D%22%20OR%20%22%5Bwarn%5D%22%20OR%20%22CRASH%22&display.page.search.mode=verbose&dispatch.sample_ratio=1&earliest=-4h%40m&latest=now&sid=1545840137.3874305))

### CODE REVIEW

File: lib/content/message/last_trip/platform_closed.ex [hunk 1]
```diff
@@ -11,7 +11,7 @@ defmodule Content.Message.LastTrip.PlatformClosed do
 
   defimpl Content.Message do
     def to_string(%Content.Message.LastTrip.PlatformClosed{}) do
-      "Platform closed"
+      "Service ended"
```
Comment (sloanelybutsurely) [inline lib/content/message/last_trip/platform_closed.ex:14]: **question:** should this be a different struct altogether? should i combine it with `%ServiceEnded{}` (which displays "Service ended for night" or "No <headsign> trains") with some sort of additional configuration?

if we use these as representations of what _kind_ of content to show, sort of the context of the message, i feel like we can keep this as-is?

File: lib/signs/utilities/last_trip.ex [hunk 2]
```diff
@@ -10,7 +10,13 @@ defmodule Signs.Utilities.LastTrip do
...
       {has_top_service_ended?, has_bottom_service_ended?} ->
         {unpacked_mz_top, unpacked_mz_bottom} = unpack_mezzanine_content(messages, source)
-        {top_source, bottom_source} = source
+        {top_source_config, bottom_source_config} = source
+
+        routes =
+          Enum.flat_map(
+            top_source_config.sources ++ bottom_source_config.sources,
+            &List.wrap(&1.routes)
+          )
```
Comment (PaulJKim) [inline lib/signs/utilities/last_trip.ex:19]: We have a helper method [here](https://github.com/mbta/realtime_signs/blob/main/lib/signs/utilities/source_config.ex#L193-L195) that should let you do something like
```suggestion
        routes = Signs.Utilities.SourceConfig.sign_routes(source)
```

File: lib/content/message/last_trip/platform_closed.ex (hunk 1 above)
Comment (PaulJKim) [reply lib/content/message/last_trip/platform_closed.ex:14]: Yeah, I agree we don't need to combine it with the `%ServiceEnded{}` struct. Like you said, the struct names just express the context of the message, which isn't changing, and whatever text they actually render to is just a minor detail imo.

File: test/signs/realtime_test.exs [hunk 3]
```diff
@@ -1406,9 +1424,20 @@ defmodule Signs.RealtimeTest do
...
     end
 
+    test "Station with routes is closed" do
+      sign = %{
+        @mezzanine_sign_with_routes
+        | tick_read: 0
+      }
+
+      expect_messages({"No Orange Line", "Service ended for night"})
+      expect_audios(canned: {"103", ["883"], :audio})
```
Comment (PaulJKim) [inline test/signs/realtime_test.exs:1434]: `883` reads out "This station is closed. Service has ended for the night.", but the ticket says it should now read "Orange Line service has ended for the night". Luckily, we already have take `882` that reads just "service has ended for the night" which can combine with takes that we have for the route names. We'll have to start passing the routes to `%Content.Audio.ServiceEnded{}` though and call a helper within its `to_params/1` implementation for the bidirectional case.

Comment (sloanelybutsurely) [reply test/signs/realtime_test.exs]: Oh nice catch! How does this look? https://github.com/mbta/realtime_signs/pull/759/commits/a799010fd5209f50bdd875b8594fec321d9f6851

Comment (PaulJKim) [reply test/signs/realtime_test.exs]: Overall seems right! But I think `routes` will always be a list though so could we get away with removing the call to `List.wrap/1`? 

I'm also a little torn on whether or not we should default to using `@service_ended`. The `Utilities.line_to_var()` helper defaults to using the take for "Train" which I think could be fine.

Comment (sloanelybutsurely) [reply test/signs/realtime_test.exs]: @PaulJKim okay, updated to remove `List.wrap` and defaulting to `@service_ended`: [`a799010..137cc1d`](https://github.com/mbta/realtime_signs/compare/a799010fd5209f50bdd875b8594fec321d9f6851..137cc1d427477a8d484281bc1d150d259030d33a?w=1)

**Review by PaulJKim — APPROVED** [review]
Looks good!

**Pull request merged**